tests/test_helpers_all_submissions.py

```diff
@@ -0,0 +1,78 @@
+"""Tests for UnauthenticatedReddit class."""
+
+from __future__ import print_function, unicode_literals
+
+import time
+
+from praw.helpers import all_submissions
+from .helper import PRAWTest, betamax
+
+
+def mock_time():
+    return 1448923624.0
+
+real_time = time.time
+
+
+class TestHelperAllSubmissions(PRAWTest):
+    def setUp(self):
+        PRAWTest.setUp(self)
+        time.time = mock_time
+
+    def tearDown(self):
+        PRAWTest.tearDown(self)
+        # make sure nose reports correct running time
+        time.time = real_time
+
+    @betamax()
+    def test_all_submissions(self):
+        try:
+            all_subs = list(all_submissions(self.r,
+                                            self.sr,
+                                            highest_timestamp=time.time(),
+                                            verbosity=3))
+
+            for i in range(len(all_subs) - 1):
+                self.assertGreaterEqual(all_subs[i].created_utc,
+                                        all_subs[i + 1].created_utc)
+
+            sr_obj = self.r.get_subreddit(self.sr)
+            all_subs_sr_object = list(all_submissions(self.r,
+                                                      sr_obj,
+                                                      verbosity=3))
+
+            def get_submissions_data(submissions_list):
+                return [(s.created_utc, s.title, s.url)
```

Comment:
1. Would it be sufficient to use the item's ID / fullname instead of these attributes? While it's unlikely that two posts would collide here, these aren't necessarily identifiers.

2. Instead of creating and comparing two lists of IDs, maybe we can zip the search results? Like:

        def assert_equal_submissions(list1, list2):
            self.assertEqual(len(list1), len(list2))
            for (item1, item2) in zip(list1, list2):
                self.assertEqual(item1.fullname, item2.fullname)

        assert_equal_submissions(all_subs, all_subs_sr_object)
        assert_equal_submissions(all_subs, reversed(all_subs_reversed))
        ...

    I think the intention is more clear this way, let me know what you think.